File: S/1582-special-positions-in-a-binary-matrix/main.go
package main

import "fmt"

func numSpecial(mat [][]int) int {
	row, col := make([]int, len(mat)), make([]int, len(mat[0]))
	for i := 0; i < len(mat); i++ {
		idx := -1
		for j := 0; j < len(mat[0]); j++ {
			if mat[i][j] == 1 {
				if idx == -1 {
					idx = j
				} else {
					idx = -1
					break
				}
			}
		}
		row[i] = idx
	}
	for i := 0; i < len(mat[0]); i++ {
		idx := -1
		for j := 0; j < len(mat); j++ {
			if mat[j][i] == 1 {
				if idx == -1 {
					idx = j
				} else {
					idx = -1
					break
				}
			}
		}
		col[i] = idx
	}

	ret := 0
	for i := 0; i < len(row); i++ {
		if row[i] == -1 {
			continue
		}
		if col[row[i]] != -1 {
			ret++
		}

	}
	return ret
}

func main() {
	fmt.Println("a")
}

File: S/1582-special-positions-in-a-binary-matrix/main.rs
impl Solution {
  pub fn num_special(mat: Vec<Vec<i32>>) -> i32 {
    let mut rows: Vec<(i32, usize)> = vec![(0, mat[0].len()); mat.len()];
    let mut cols: Vec<i32> = vec![0; mat[0].len()];
    for r in 0..mat.len() {
      let mut cc: usize = mat[0].len();
      for c in 0..mat[0].len() {
        rows[r].0 += mat[r][c];
        cols[c] += mat[r][c];
        if mat[r][c] == 1 {
          cc = c;
        }
      }
      if rows[r].0 == 1 {
        rows[r].1 = cc;
      }
    }

    rows.iter().fold(0, |sum, row| {
      if row.0 != 1 {
        sum
      } else {
        sum
          + if row.1 == mat[0].len() || cols[row.1] != 1 {
            0
          } else {
            1
          }
      }
    })
  }
}
